streams/src/main/java/org/apache/kafka/streams/state/internals/MergedSortedCacheWindowStoreIterator.java

```diff
@@ -22,17 +22,22 @@
 import org.apache.kafka.streams.state.StateSerdes;
 import org.apache.kafka.streams.state.WindowStoreIterator;
 
+import static org.apache.kafka.streams.state.internals.SegmentedCacheFunction.bytesFromCacheKey;
+
 /**
  * Merges two iterators. Assumes each of them is sorted by key
  *
  * @param <V>
  */
-class MergedSortedCacheWindowStoreIterator<V> extends AbstractMergedSortedCacheStoreIterator<Long, Long, V> implements WindowStoreIterator<V> {
+class MergedSortedCacheWindowStoreIterator<V> extends AbstractMergedSortedCacheStoreIterator<Long, Long, V, byte[]> implements WindowStoreIterator<V> {
+
+    private final StateSerdes<Long, V> serdes;
```

Comment:
AbstractMergedSortedCacheStoreIterator doesn't need the serdes, since it delegates all the serialization / deserialization through its abstract methods. The main motivation was this bug https://github.com/apache/kafka/pull/3016 which was caused by calling serdes directly within the abstract class.